packages/reactive-element/src/reactive-element.ts

```diff
@@ -1038,19 +1038,32 @@ export abstract class ReactiveElement
    */
   private __saveInstanceProperties() {
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    const instanceProperties: Array<[keyof this, any]> = [];
+    const instanceProperties = new Map<keyof this, this[keyof this]>();
     const elementProperties = (this.constructor as typeof ReactiveElement)
       .elementProperties;
     for (const p of elementProperties.keys() as IterableIterator<keyof this>) {
       if (this.hasOwnProperty(p)) {
-        instanceProperties.push([p, this[p]]);
+        instanceProperties.set(p, this[p]);
         delete this[p];
       }
     }
-    if (instanceProperties.length) {
+    if (elementProperties.size > 0) {
       queueMicrotask(() => {
-        for (const [p, value] of instanceProperties) {
-          this[p] = value;
+        for (const [p, options] of elementProperties) {
+          if (instanceProperties.has(p as keyof this)) {
+            this[p as keyof this] = instanceProperties.get(p as keyof this)!;
+          } else if (
+            this[p as keyof this] !== undefined &&
+            !this._$changedProperties.has(p)
+          ) {
+            // Trigger initial value reflection and populate the initial
+            // changedProperties map, but only for the case of experimental
+            // decorators on accessors, which will not have already populated
+            // the changedProperties map. We skip `undefined` values assuming
+            // those aren't initial values - a small difference from
+            // experimental decorators on fields.
+            this.requestUpdate(p, undefined, options);
```

Comment:
Does this trigger a second render, or is this still before the initial render? Do we have a test of that?

Reply:
There are tests that count the number of updates - even the number of `hasChanged()` calls, which happen every requestUpdate instead of every update. This behaves exactly the same in all the cases we currently test. We may be missing coverage for something like a property that reflects, is initialized to `undefined` _and_ has an attribute converter that converts `undefined` to a value rather than removing the attribute.